other-docs/getting-started/custom-modules.md

```diff
@@ -0,0 +1,165 @@
+    ]
+}
+```
+
+After adding this, run `composer dump-autoload` to regenerate the autoload files for your project.
+
+
+### Registering Your Module
+
+Within your module's entrypoint file, your module needs to register itself with the Platform core. To do this, you need to add an action on `hm-platform.modules.init` which calls `HM\Platform\Module::register`. This function call registers the module as well as any default configuration and a callback for the module.
+
+A basic registration call looks like:
+
+```php
+use function HM\Platform\register_module;
+
+add_action( 'hm-platform.modules.init', function () {
+	register_module(
+		// Module ID:
+		'your-module',
+
+		// Module directory:
+		__DIR__,
+
+		// Human-readable name:
+		'Your Module',
+
+		// Default configuration, as an array:
+		[
+			// `enabled` is a special configuration key, which sets whether the module is
+			// enabled by default or not.
+			'enabled' => true,
+		],
+
+		// Function to call if the module is enabled.
+		__NAMESPACE__ . '\\bootstrap'
+	);
+} );
+```
+
+We recommend using a function-exists-guard to only register your action if running in a Platform context, as the autoloader can also be used by static analysis and other tools including PHPUnit. Your function-exists-guard should look like:
+
+```php
+// Don't self-initialize if this is not a Platform execution.
+if ( ! function_exists( 'add_action' ) ) {
+	return;
+}
+```
+
+Your entrypoint file can declare constants or load in other files as necessary to load functions and classes. A typical entrypoint file looks like:
+
+```php
+<?php
+
+namespace YourProject;
+
+use function HM\Platform\register_module;
+
+const DIRECTORY = __DIR__;
+
+// Load in namespaced-functions.
+require_once __DIR__ . '/inc/namespace.php';
+
+// Don't self-initialize if this is not a Platform execution.
+if ( ! function_exists( 'add_action' ) ) {
+	return;
+}
+
+add_action( 'hm-platform.modules.init', function () {
+	register_module(
+		'your-module',
+		DIRECTORY,
+		'Your Module',
+		[
+			'enabled' => true,
+		],
+		__NAMESPACE__ . '\\bootstrap'
+	);
+} );
+```
+
+
+### Module Settings
+
+When registering your module, you pass various parameters to `Module::register`. These are used to control the behaviour of your module.
+
+Your module's directory is used for automated loading and parsing of documentation. The Documentation module automatically looks for a `docs` subdirectory within this directory and parses Markdown files from this.
+
+The default configuration passed in integrates with the [configuration system](configuration.md), and can be used to configure your module on different environments, or to provide an easy way to enable and disable specific functionality. Within your module, you can use `HM\Platform\get_config()` to retrieve the resolved configuration; your module configuration can be accessed as `get_config( $id )`, where `$id` is the same ID you pass to the module registration.
```

Comment:
`get_config()` doesn't take any params. The module config can be accessed from the module object though using `$module->get_settings()`. Should perhaps be `$module->get_config()` ...